src/test/java/com/emc/mongoose/storage/driver/pravega/util/docker/MongooseContainer.java

```diff
@@ -0,0 +1,206 @@
+	private final int durationLimitSeconds;
+	private String testContainerId = null;
+	private long duration = -1;
+	private String stdOutput = null;
+	private int containerExitCode = -1;
+
+	public long getDuration() {
+		return duration;
+	}
+
+	public String getStdOutput() {
+		return stdOutput;
+	}
+
+	private int getContainerExitCode() {
+		return containerExitCode;
+	}
+
+	public MongooseContainer(final List<String> configArgs, final int durationLimitSeconds) {
+		this.durationLimitSeconds = durationLimitSeconds;
+		this.dockerClient = DockerClientBuilder.getInstance().build();
+		final File dockerBuildFile = Paths
+				.get(BASE_DIR, "docker", "Dockerfile")
+				.toFile();
+		LOG.info("Build mongoose image w/ Pravega support using the dockerfile " + dockerBuildFile);
+		final BuildImageResultCallback buildImageResultCallback = new BuildImageResultCallback();
+		this.dockerClient
+				.buildImageCmd()
+				.withBaseDirectory(new File(BASE_DIR))
+				.withDockerfile(dockerBuildFile)
+				.withBuildArg("MONGOOSE_VERSION", APP_VERSION)
+				.withPull(true)
+				.withTags(Collections.singleton("emcmongoose/mongoose-storage-driver-pravega:testing"))
+				.exec(buildImageResultCallback);
+		final String testingImageId = buildImageResultCallback.awaitImageId();
+		LOG.info("Build mongoose testing image id: " + testingImageId);
+		this.configArgs.add("--output-metrics-trace-persist=true");
+		this.configArgs.add("--storage-net-node-port=" + PravegaNodeContainer.PORT);
+		for (final String configArg : configArgs) {
+			if (configArg.startsWith("--run-scenario=")) {
+				final String scenarioPathStr = configArg.substring("--run-scenario=".length());
+				if (scenarioPathStr.startsWith(HOST_SHARE_PATH.toString())) {
+					this.configArgs.add(
+							"--run-scenario=" + CONTAINER_SHARE_PATH
+									+ scenarioPathStr.substring(HOST_SHARE_PATH.toString().length())
+					);
+				} else {
+					this.configArgs.add(
+							"--run-scenario=" + CONTAINER_SHARE_PATH + scenarioPathStr
+					);
+				}
+			} else {
+				this.configArgs.add(configArg);
+			}
+		}
+		LOG.info("Mongoose test container arguments: " + Arrays.toString(configArgs.toArray()));
+		final Volume volumeShare = new Volume(CONTAINER_SHARE_PATH);
+		final Volume volumeLog = new Volume(CONTAINER_LOG_PATH);
+		final Bind[] binds = new Bind[]{
+				new Bind(HOST_SHARE_PATH.toString(), volumeShare),
+				new Bind(HOST_LOG_PATH.toString(), volumeLog),
+		};
+		// put the environment variables into the container
+		final Map<String, String> envMap = System.getenv();
+		final String[] env = envMap.keySet().toArray(new String[envMap.size()]);
+		for (int i = 0; i < env.length; i++) {
+			if ("PATH".equals(env[i])) {
+				env[i] = env[i] + "=" + envMap.get(env[i]) + ":/bin";
+			} else {
+				env[i] = env[i] + "=" + envMap.get(env[i]);
+			}
+		}
+		final CreateContainerResponse container = dockerClient
+				.createContainerCmd(testingImageId)
+				.withName("mongoose")
```

Comment:
should be `mongoose-storage-driver-pravega`
https://hub.docker.com/r/emcmongoose/mongoose-storage-driver-pravega/

Reply:
@akurilov I had done it using an [example](https://github.com/emc-mongoose/mongoose-storage-driver-hdfs/blob/e2d977c809022d05c1cde39324fa5d56f1997af0/src/test/java/com/emc/mongoose/storage/driver/hdfs/util/docker/MongooseContainer.java#L152) hdfs driver

Could you explain why in hdfs used name `mongoose` but in pravega, I should be used `mongoose-storage-driver-pravega`?

Reply:
I was wrong. We build the testing image right here in this code above, so the name doesn't matter.